### PR TITLE
Add html-designer example to perform hot reload of html render

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This change adds a script to improve the html edit workflow.

### CODE REVIEW

**Comment by TristanCacqueray** [comment]
There are a bunch of TODOs remaining, but the core logic seems to work. Though this is clearly overkill, and using an existing tool might be better. I guess this is good learning material too :)

Perhaps that could be moved to a standalone project?

**Comment by elmiko** [comment]
this is very cool, thanks Tristan!

> Perhaps that could be moved to a standalone project?

i have been thinking about whether to add the server functionality into this tool, and the more i hack on it, the more i like the simplicity of tool that just takes a path as input and prints some output to stdout. it feels very unix-esque.

File: tools/html-designer.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2022 Red Hat
+// SPDX-License-Identifier: Apache-2.0
```
Comment (elmiko) [inline tools/html-designer.rs]: this repo is under a gpl-3-or-later license, any objection to changing this?

Comment (TristanCacqueray) [reply tools/html-designer.rs]: sure.

**Comment by elmiko** [comment]
i'm going to play around with this a little but more, but as i add more test data to [testdata/must-gather-valid](https://github.com/elmiko/camgi.rs/tree/devel/testdata/must-gather-valid/sample-openshift-release) i think it would be cool to have the designer just look at that directory.

**Comment by TristanCacqueray** [comment]
The designer currently does two things:

- watch for [src/html.rs](https://github.com/TristanCacqueray/camgi.rs/blob/ce0bbe0c9394a735c8fb836a53173af1186784c8/tools/html-designer.rs#L98-L100) and run [cargo run](https://github.com/TristanCacqueray/camgi.rs/blob/html-designer/tools/html-designer.rs#L109-L122) when the file change.
- watch for [./target/html/index.html](https://github.com/TristanCacqueray/camgi.rs/blob/ce0bbe0c9394a735c8fb836a53173af1186784c8/tools/html-designer.rs#L21) and reload the web page when the file change: https://github.com/TristanCacqueray/camgi.rs/blob/ce0bbe0c9394a735c8fb836a53173af1186784c8/tools/html-designer.rs#L124-L133

I guess we could make the designer takes some argument to toggle the behavior.

**Comment by TristanCacqueray** [comment]
Withh 3e7e421, you can now pass the html file to watch on the designer command line.

**Comment by elmiko** [comment]
ooh, i think i misunderstood how the patch works. i'll study a little deeper.

File: src/main.rs [hunk 2]
```diff
@@ -26,9 +26,19 @@ struct Cli {
 fn main() -> Result<()> {
     let cli = Cli::parse();
 
-    let mg = MustGather::from(cli.path)?;
+    if cli.path == "demo" {
+        // Special case to render a demo report
+        let mg =
+            MustGather::from("testdata/must-gather-valid/sample-openshift-release/".to_string())?;
```
Comment (elmiko) [inline src/main.rs:32]: :+1:

**Review by elmiko — COMMENTED** [review]
i was trying to test this out while making some changes to html.rs, but when i run it i'm getting this error:
```
$ cargo run --example html-designer
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/examples/html-designer`
thread 'main' panicked at 'Can't read the file: Os { code: 2, kind: NotFound, message: "No such file or directory" }', tools/html-designer.rs:61:63
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

i'm running it from the root of the repo and src/html.rs is there. did i miss some configuration?

**Comment by elmiko** [comment]
i guess i need to run `cargo run demo` first?

**Comment by elmiko** [comment]
it's weird, looking at the code i would expect it to automatically "do all the things", but for some reason it's not behaving that way for me. i had to manually run `cargo run demo` the first time, and then it didn't re-run the command for me.

**Comment by TristanCacqueray** [comment]
That's correct, the last two commits should now take care of that, so that it does all the things.

**Comment by elmiko** [comment]
cool!
i'll give it another try tonight , thanks =)

**Comment by elmiko** [comment]
this is working for me, should i merge it @TristanCacqueray or did you have some other ideas to add?

**Comment by TristanCacqueray** [comment]
I think this is good to go, thanks for the feedback!

**Pull request merged**